crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -305,6 +310,75 @@ impl EntityTopicId {
         }
         .map(|parent_id| EntityTopicId(format!("device/{parent_id}//")))
     }
+
+    /// If `self` is a device topic id, return a service topic id under this
+    /// device.
+    ///
+    /// The device topic id must be in a format: "device/DEVICE_NAME//"; if not,
+    /// `None` will be returned.
+    pub fn to_service_topic_id(&self, service_name: &str) -> Option<ServiceTopicId> {
+        if let ["device", device_name, "", ""] = self.0.split('/').collect::<Vec<&str>>()[..] {
+            return Some(ServiceTopicId(EntityTopicId(format!(
+                "device/{device_name}/service/{service_name}"
+            ))));
+        }
+        None
+    }
+
+    /// Returns an array of all segments of this entity topic.
+    pub fn segments(&self) -> [&str; ENTITY_ID_SEGMENTS] {
```

Comment:
I would really avoid to have this method `pub`: from the API a topic Id is atomic.

Reply:
Would having this be `pub(crate)` be acceptable?